### PR TITLE
Adds support for GitHub, Azure AD, Google and SAML auth to the UI

### PR DESCRIPTION
**Pull request by mabashian**
##### SUMMARY
link #4204 

This is a fork of the branch used for #8619.  This PR should not merge until that one does.  Once that happens I'll rebase this branch.

Screen Shot 2020-12-01 at 2 47 32 PM

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - UI

### CODE REVIEW

File: awx/ui_next/src/setupProxy.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ const TARGET = process.env.TARGET || 'https://localhost:8043';
 
 module.exports = app => {
   app.use(
-    createProxyMiddleware(['/api', '/websocket'], {
+    createProxyMiddleware(['/api', '/websocket', '/sso'], {
```
Comment (jakemcdermott) [inline awx/ui_next/src/setupProxy.js:7]: :+1:

**Review by jakemcdermott — APPROVED** [review]

**Comment by unlikelyzero** [comment]
@one-t and @tiagodread will verify this manually and take a look at what's involved in using https://github.com/lirantal/cypress-social-logins

1. Manually configure and verify functionality.
2. Explore automation options in the closedsource repo.

Maybe this? https://github.com/cypress-io/cypress-example-recipes/tree/master/examples/logging-in__single-sign-on

**Comment by mabashian** [comment]
@unlikelyzero @tiagodread @one-t this should be ready for test now that #8619 has merged

**Comment by unlikelyzero** [comment]
@mabashian , @one-t and @tiagodread have tested this with Okta SAML and used our semi-automated internal testing infrastructure. How and _when_ we test Oauth automatically is still not decided, but we've got stubbed coverage in 5783

I think this would be the perfect opportunity for the community to test the other SSO Options.

**Comment by mabashian** [comment]
@unlikelyzero @tiagodread @one-t so we merging or y'all wanna wait?

**Comment by ryanpetrello** [comment]
regate

**Pull request merged**